deployment/tfhub/object-detection-url.ipynb

```diff
@@ -0,0 +1,643 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CtIgsA4aB2ga"
+      },
+      "source": [
+        "## 2. Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7SuugL_lOuaN"
+      },
+      "outputs": [],
+      "source": [
+        "import concurrent.futures\n",
+        "import os\n",
+        "import pandas as pd\n",
+        "import requests\n",
+        "import tempfile\n",
+        "import tensorflow as tf\n",
+        "import tensorflow_hub as hub\n",
+        "import time\n",
+        "import urllib.request\n",
+        "\n",
+        "from bs4 import BeautifulSoup\n",
+        "from PIL import Image, ImageOps\n",
+        "from verta import Client\n",
+        "from verta.endpoint.autoscaling import Autoscaling\n",
+        "from verta.endpoint.autoscaling.metrics import CpuUtilizationTarget, MemoryUtilizationTarget, RequestsPerWorkerTarget\n",
+        "from verta.endpoint.resources import Resources\n",
+        "from verta.endpoint.update import DirectUpdateStrategy\n",
+        "from verta.environment import Python\n",
+        "from verta.registry import VertaModelBase, verify_io\n",
+        "from verta.utils import ModelAPI"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "LB20pSYuXcCD"
+      },
+      "source": [
+        "## 3. Verta Set Up"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "hGwbnFGlXgJh"
+      },
+      "outputs": [],
+      "source": [
+        "os.environ['VERTA_HOST'] = ''\n",
+        "os.environ['VERTA_EMAIL'] = ''\n",
+        "os.environ['VERTA_DEV_KEY'] = ''"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "wVcbn1_3Xhtm",
+        "outputId": "ad2bb3bf-32ce-499e-8b2c-a2efa6a7c45a"
+      },
+      "outputs": [],
+      "source": [
+        "client = Client(os.environ['VERTA_HOST'])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5bBHXvKnX_Je"
+      },
+      "source": [
+        "## 4. Model Class"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "o3YNlqVMEi6c"
+      },
+      "outputs": [],
+      "source": [
+        "class DetectObject(VertaModelBase):\n",
+        "    def __init__(self, artifacts=None):\n",
+        "        module_handle = 'https://tfhub.dev/google/openimages_v4/ssd/mobilenet_v2/1'\n",
+        "        self.detector = hub.load(module_handle).signatures['default']\n",
+        "    \n",
+        "    def handle_img(self, data, width=640, height=480):\n",
+        "        _, path = tempfile.mkstemp(suffix='.jpg')\n",
+        "        file_name = data[0].split('/')[-1]\n",
```

Comment:
Is there a reason to use `.split('/')[-1]` instead of `os.path.basename()`?

Reply:
Actually, isn't `data[0]` already the filename? 🤔

Reply:
I was updating to `os.path.basename()` instead of `split()`, but as `predict()` was refactored to:
```
def predict(data):
    file_name, url  = data
```

we don't need any of them here. The `os.path.basename()` command is being used before calling `predict()`, to pass the file name as an argument. :+1: